File: src/test/java/de/alpharogroup/file/zip/SevenZipTest.java
/**
 * The MIT License
 *
 * Copyright (C) 2020 Asterios Raptis
 *
 * Permission is hereby granted, free of charge, to any person obtaining
 * a copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so, subject to
 * the following conditions:
 *
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
 * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
package de.alpharogroup.file.zip;

import static org.junit.jupiter.api.Assertions.*;

import java.io.File;
import java.io.IOException;

import org.junit.jupiter.api.Test;

import io.github.astrapi69.file.create.DirectoryFactory;
import io.github.astrapi69.file.create.FileFactory;
import io.github.astrapi69.file.delete.DeleteFileExtensions;
import io.github.astrapi69.file.search.PathFinder;
import io.github.astrapi69.file.write.StoreFileExtensions;

class SevenZipTest
{

	@Test
	void zipFiles() throws IOException
	{
		File testFile1;
		File testFile2;
		File testFile3;
		File destination;
		File sevenZipFile;

		destination = new File(PathFinder.getSrcTestResourcesDir(), "/tmp");
		DirectoryFactory.newDirectory(destination);
		sevenZipFile = new File(destination, "testzip.7z");
		FileFactory.newFile(sevenZipFile);

		testFile1 = new File(PathFinder.getSrcTestResourcesDir(), "testZip1.txt");
		testFile2 = new File(PathFinder.getSrcTestResourcesDir(), "testZip2.txt");
		testFile3 = new File(PathFinder.getSrcTestResourcesDir(), "testZip3.txt");

		StoreFileExtensions.toFile(testFile1, "Its a beautifull day!!!");
		StoreFileExtensions.toFile(testFile2, "Its a beautifull evening!!!");
		StoreFileExtensions.toFile(testFile3, "Its a beautifull night!!!");

		SevenZip.zipFiles(sevenZipFile, "", testFile1, testFile2, testFile3);

		assertTrue(sevenZipFile.exists());
		// clean up
		DeleteFileExtensions.delete(sevenZipFile);
		DeleteFileExtensions.delete(testFile1);
		DeleteFileExtensions.delete(testFile2);
		DeleteFileExtensions.delete(testFile3);
		DeleteFileExtensions.delete(destination);
	}

	@Test
	void testExtract() throws IOException
	{
		File testFile1;
		File testFile2;
		File testFile3;
		File destination;
		File sevenZipFile;

		destination = new File(PathFinder.getSrcTestResourcesDir(), "/tmp");
		DirectoryFactory.newDirectory(destination);
		sevenZipFile = new File(destination, "testzip.7z");
		FileFactory.newFile(sevenZipFile);

		testFile1 = new File(PathFinder.getSrcTestResourcesDir(), "testZip1.txt");
		testFile2 = new File(PathFinder.getSrcTestResourcesDir(), "testZip2.txt");
		testFile3 = new File(PathFinder.getSrcTestResourcesDir(), "testZip3.txt");

		StoreFileExtensions.toFile(testFile1, "Its a beautifull day!!!");
		StoreFileExtensions.toFile(testFile2, "Its a beautifull evening!!!");
		StoreFileExtensions.toFile(testFile3, "Its a beautifull night!!!");

		assertTrue(testFile1.exists());
		assertTrue(testFile2.exists());
		assertTrue(testFile3.exists());

		SevenZip.zipFiles(sevenZipFile, "", testFile1, testFile2, testFile3);

		DeleteFileExtensions.delete(testFile1);
		DeleteFileExtensions.delete(testFile2);
		DeleteFileExtensions.delete(testFile3);

		SevenZip.extract(sevenZipFile, destination);

		testFile1 = new File(destination, "testZip1.txt");
		testFile2 = new File(destination, "testZip2.txt");
		testFile3 = new File(destination, "testZip3.txt");
		assertTrue(testFile1.exists());
		assertTrue(testFile2.exists());
		assertTrue(testFile3.exists());
		// clean up
		DeleteFileExtensions.delete(sevenZipFile);
		DeleteFileExtensions.delete(testFile1);
		DeleteFileExtensions.delete(testFile2);
		DeleteFileExtensions.delete(testFile3);
		DeleteFileExtensions.delete(destination);
	}

}

File: src/test/kotlin/io/github/astrapi69/file/sevenzip/SevenZipTest.kt
package io.github.astrapi69.file.sevenzip

import de.alpharogroup.file.zip.SevenZip
import java.io.*
import java.nio.file.Files
import org.junit.jupiter.api.*

class SevenZipTest {

  private lateinit var tempDir: File
  private lateinit var tempZipFile: File
  private val testPassword = "password".toCharArray()

  @BeforeEach
  fun setup() {
    tempDir = Files.createTempDirectory("testDir").toFile()
    tempZipFile = Files.createTempFile("test", ".7z").toFile()
  }

  @AfterEach
  fun teardown() {
    tempDir.deleteRecursively()
    if (tempZipFile.exists()) {
      tempZipFile.delete()
    }
  }

  @Test
  fun `test zipFiles`() {
    val file1 = File(tempDir, "file1.txt").apply { writeText("Hello World 1") }
    val file2 = File(tempDir, "file2.txt").apply { writeText("Hello World 2") }

    SevenZip.zipFiles(tempZipFile, tempDir.absolutePath, file1, file2)

    Assertions.assertTrue(tempZipFile.exists())
    Assertions.assertTrue(tempZipFile.length() > 0)
  }

  @Test
  fun `test extract without password`() {
    val file1 = File(tempDir, "file1.txt").apply { writeText("Hello World 1") }
    val file2 = File(tempDir, "file2.txt").apply { writeText("Hello World 2") }

    SevenZip.zipFiles(tempZipFile, file1, file2)

    val extractDir = Files.createTempDirectory("extractDir").toFile()
    SevenZip.extract(tempZipFile, extractDir)

    Assertions.assertTrue(File(extractDir, "file1.txt").exists())
    Assertions.assertTrue(File(extractDir, "file2.txt").exists())

    extractDir.deleteRecursively()
  }
}
